editor/src/core/shared/github/helpers.ts

```diff
@@ -178,370 +153,31 @@ export function repositoryEntry(
-            ),
-            updateGithubSettings(
-              projectGithubSettings(
-                persistentModel.githubSettings.targetRepository,
-                responseBody.newCommit,
-                responseBody.branchName,
-                responseBody.newCommit,
-                true,
-              ),
-            ),
-            updateBranchContents(persistentModel.projectContents),
-            showToast(notice(`Saved to branch ${responseBody.branchName}.`, 'SUCCESS')),
-          ],
-          'everyone',
-        )
-
-        // refresh the branches after the content was saved
-        if (persistentModel.githubSettings.targetRepository != null) {
-          void dispatchPromiseActions(
-            dispatch,
-            getBranchesForGithubRepository(
-              dispatch,
-              persistentModel.githubSettings.targetRepository,
-            ),
-          )
-        }
-        break
-      default:
-        const _exhaustiveCheck: never = responseBody
-        throw new Error(`Github: Unhandled response body ${JSON.stringify(responseBody)}`)
-    }
-  } else {
-    dispatch(
-      [
-        showToast(
-          notice(`Github: Unexpected status returned from endpoint: ${response.status}`, 'ERROR'),
-        ),
-      ],
-      'everyone',
-    )
-  }
-  dispatch([updateGithubOperations(operation, 'remove')], 'everyone')
-}
-
-export async function getBranchesForGithubRepository(
-  dispatch: EditorDispatch,
-  githubRepo: GithubRepo,
+  logic: () => Promise<Array<EditorAction>>,
 ): Promise<Array<EditorAction>> {
-  const operation: GithubOperation = { name: 'listBranches' }
-
-  dispatch([updateGithubOperations(operation, 'add')], 'everyone')
-
+  let result: Array<EditorAction> = []
   try {
-    const url = urljoin(
-      UTOPIA_BACKEND,
-      'github',
-      'branches',
-      githubRepo.owner,
-      githubRepo.repository,
-    )
-
-    const response = await fetch(url, {
-      method: 'GET',
-      credentials: 'include',
-      headers: HEADERS,
-      mode: MODE,
-    })
-
-    if (response.ok) {
-      const responseBody: GetBranchesResponse = await response.json()
-
-      switch (responseBody.type) {
-        case 'FAILURE':
-          return [
-            showToast(
-              notice(`Error when listing branches: ${responseBody.failureReason}`, 'ERROR'),
-            ),
-          ]
-        case 'SUCCESS':
-          return [updateGithubData({ branches: responseBody.branches })]
-        default:
-          const _exhaustiveCheck: never = responseBody
-          throw new Error(`Unhandled response body ${JSON.stringify(responseBody)}`)
-      }
-    } else {
-      return [
-        showToast(
-          notice(`Github: Unexpected status returned from endpoint: ${response.status}`, 'ERROR'),
-        ),
-      ]
-    }
+    dispatch([updateGithubOperations(operation, 'add')], 'everyone')
+    await new Promise((res) => setTimeout(res, 1000))
```

Comment:
what are we really awaiting here?

Reply:
debug remnants I was sure I did not push 🙈

Reply:
oh :D